src/custom/logic.rs

```diff
@@ -0,0 +1,169 @@
+use hugr::{
+    extension::{simple_op::MakeExtensionOp, ExtensionId},
+    ops::{CustomOp, Value},
+    std_extensions::logic::{self, ConcreteLogicOp, NaryLogic},
+    types::{CustomType, SumType},
+    HugrView,
+};
+use inkwell::{types::BasicTypeEnum, IntPredicate};
+
+use crate::{
+    emit::{emit_value, func::EmitFuncContext, EmitOp, EmitOpArgs},
+    sum::LLVMSumValue,
+    types::TypingSession,
+};
+
+use super::{CodegenExtension, CodegenExtsMap};
+use anyhow::{anyhow, Result};
+
+struct LogicOpEmitter<'c, 'd, H>(&'d mut EmitFuncContext<'c, H>);
+
+impl<'c, H: HugrView> EmitOp<'c, CustomOp, H> for LogicOpEmitter<'c, '_, H> {
+    fn emit(&mut self, args: EmitOpArgs<'c, CustomOp, H>) -> Result<()> {
+        let lot = ConcreteLogicOp::from_optype(&args.node().generalise()).ok_or(anyhow!(
+            "LogicOpEmitter: from_optype_failed: {:?}",
+            args.node().as_ref()
+        ))?;
+        let builder = self.0.builder();
+        // Turn bool sum inputs into i1's
+        let mut inputs = vec![];
+        for inp in args.inputs {
+            let bool_ty = self.0.llvm_sum_type(SumType::Unit { size: 2 })?;
+            let bool_val = LLVMSumValue::try_new(inp, bool_ty)?;
+            inputs.push(bool_val.build_get_tag(builder)?);
```

Comment:
I think this means that true = 0 and false = 1. I wonder if we can have it the other round. I guess the same logic applies to sum types for CFG branches, so not for this PR...

Reply:
Yeah I think that's the case. Other functions like `emit_icmp` use the same encoding